cognite_toolkit/cdf_tk/templates/_templates.py

```diff
@@ -377,74 +416,107 @@ def sort_key(p: Path) -> int:
-                and re.match(FunctionLoader.filename_pattern, filepath.stem)
-            ):
-                if not printed_function_warning and sys.version_info >= (3, 12):
-                    print(
-                        "      [bold yellow]WARNING:[/] The functions API does not support Python 3.12. "
-                        "It is recommended that you use Python 3.11 or 3.10 to develop functions locally."
+                validate(content, destination, filepath, modules_by_variables)
+                source_by_build_path[destination] = filepath
+                # If we have a function definition, we want to process the directory.
+                if (
+                    resource_folder == "functions"
+                    and filepath.suffix.lower() == ".yaml"
+                    and re.match(FunctionLoader.filename_pattern, filepath.stem)
+                ):
+                    if not printed_function_warning and sys.version_info >= (3, 12):
+                        print(
+                            "      [bold yellow]WARNING:[/] The functions API does not support Python 3.12. "
+                            "It is recommended that you use Python 3.11 or 3.10 to develop functions locally."
+                        )
+                        printed_function_warning = True
+                    process_function_directory(
+                        yaml_source_path=filepath,
+                        yaml_dest_path=destination,
+                        module_dir=module_dir,
+                        build_dir=build_dir,
+                        common_code_dir=Path(project_config_dir / environment.common_function_code),
+                        verbose=verbose,
                     )
-                    printed_function_warning = True
-
-                process_function_directory(
-                    yaml_source_path=filepath,
-                    yaml_dest_path=destination,
-                    module_dir=module_dir,
-                    build_dir=build_dir,
-                    common_code_dir=Path(project_config_dir / environment.common_function_code),
-                )
+                    all_files[resource_folder]["other_files"] = []
+                if resource_folder == "files":
+                    process_files_directory(
+                        files=all_files[resource_folder]["other_files"],
+                        yaml_dest_path=destination,
+                        module_dir=module_dir,
+                        build_dir=build_dir,
+                        verbose=verbose,
+                    )
+                    all_files[resource_folder]["other_files"] = []
+
+            if resource_folder == "timeseries_datapoints":
+                # Process all csv files
+                for filepath in all_files["timeseries_datapoints"]["other_files"]:
+                    if filepath.suffix.lower() == ".csv":
+                        # Special case for timeseries datapoints, we want to timeshift datapoints
```

Comment:
If we only process `.csv` files here, I would rather do an early exit and remove one indent from the block underneath to make that clearer

```suggestion
                    if filepath.suffix.lower() != ".csv":
                        continue 
                     # Special case for timeseries datapoints, we want to timeshift datapoints
```